dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -24,6 +25,34 @@ def find_toxicity(role, text, locale)
         comprehend_response = AichatComprehendHelper.get_toxicity(text, locale)
         return {text: text, blocked_by: 'comprehend', details: comprehend_response} if comprehend_response && comprehend_response[:toxicity] > threshold
       end
+
+      if openai_enabled?(role)
+        details = openai_safety_check(text)
+        return {text: text, blocked_by: 'openai', details: details} if details
+      end
+    end
+
+    private def openai_safety_check(text)
+      system_prompt_option = get_openai_system_prompt
+      safety_system_prompt = AichatSystemPromptHelper.get_system_prompt(system_prompt_option)
+      details = nil
+      # Try twice in case of network errors or model not correctly following directions and
+      # replying with something other valid expected output.
+      Retryable.retryable(tries: 2) do
+        openai_response = OpenaiChatHelper.request_safety_check(text, safety_system_prompt)
+        # For now, we only have one option for safety simple prompt.
+        if system_prompt_option == 'simple'
```

Comment:
I'd suggest simplifying here and baking in the prompt, and changing it when it needs changing. Having "dead code paths" doesn't seem like it improves things, and adds confusion in the future. The only semi-universal software engineering principle I know of is: when given the option, choose the simpler path.

Its probably actually faster and easier to updating a hardcoded string prompt than having to read the code and figure out how to add a key, and its equally as hardcoded either way.

Reply:
Sure - this makes sense. When I initially wrote this, I wasn't sure if we were going to implement your prompt from testing or the few-shot prompt, but I understand now we'll go with the tested prompt first. Thanks!